src/bmi.jl

```diff
@@ -59,11 +59,19 @@ end
 # Read into memory for now with read, to avoid locking the file, since it mmaps otherwise.
 # We could pass Mmap.mmap(path) ourselves and make sure it gets closed, since Arrow.Table
 # does not have an io handle to close.
-read_table(entry::AbstractString) = Arrow.Table(read(entry))
-
-function read_table(entry)
-    @assert Tables.istable(entry)
-    return entry
+_read_table(entry::AbstractString) = Arrow.Table(read(entry))
+_read_table(entry) = entry
+
+function read_table(entry; schema = nothing)
+    table = _read_table(entry)
+    @assert Tables.istable(table)
+    if !isnothing(schema)
+        sv = schema()
+        validate(Tables.schema(table), sv)
+        R = Legolas.record_type(sv)
+        foreach(R, Tables.rows(table))  # construct each row
```

Comment:
Do you know roughly how long this takes for the national schematization? Especially the forcing can contain millions of rows, and sometimes only a fraction of them is used for short calculations.

Also, the result of this loop is not kept, that means that we only get errors, but any changes like [`clamp` in the Legolas tour](https://github.com/beacon-biosignals/Legolas.jl/blob/b22e0ce2ed7e2846c1d62b893e0e1bf63eb1f509/examples/tour.jl#L151-L155) is never done, or not?

Reply:
That's correct, at the moment I only check the types and for empty strings. And I rather throw on error than clamp some values, but that's up for debate.

Reply:
Ok yeah right now we don't use something like clamp, and I can't directly think of a use case where it would be better than an error, so let's keep it like this.

Any idea about the performance of this line?